LibVLCSharp.Forms/Shared/PlaybackControls.xaml.cs

```diff
@@ -0,0 +1,1384 @@
+                    }
+
+                    var videoTrack = mediaTrack.Value.Data.Video;
+                    var videoWidth = videoTrack.Width;
+                    var videoHeight = videoTrack.Height;
+                    if (videoWidth == 0 || videoHeight == 0)
+                    {
+                        return;
+                    }
+
+                    var sarDen = videoTrack.SarDen;
+                    var sarNum = videoTrack.SarNum;
+                    if (sarNum != sarDen)
+                    {
+                        videoWidth = videoWidth * sarNum / sarDen;
+                    }
+
+                    var var = (double)videoWidth / videoHeight;
+                    var scalingFactor = Device.Info.ScalingFactor;
+                    var screenWidth = videoView.Width * scalingFactor;
+                    var screenHeight = videoView.Height * scalingFactor;
+                    var screenar = screenWidth / screenHeight;
+                    mediaPlayer.Scale = (float)(screenar >= var ? screenWidth / videoWidth : screenHeight / videoHeight);
+                }
+                else
+                {
+                    mediaPlayer.Scale = 0;
+                }
+            }
+            catch (Exception ex)
+            {
+                ShowErrorMessageBox(ex);
+            }
+        }
+
+        private void SeekBar_ValueChanged(object sender, ValueChangedEventArgs e)
+        {
+            _ = FadeInAsync();
+
+            UpdateTime();
+            SeekBarTimerEnabled = true;
+            SeekBarTimer.Change(TimeSpan.FromSeconds(1), TimeSpan.FromMilliseconds(-1));
+        }
+
+        private void UpdateMediaPlayerPosition()
+        {
+            try
+            {
+                var mediaPlayer = MediaPlayer;
+                if (mediaPlayer != null)
+                {
+                    mediaPlayer.Position = (float)(SeekBar.Value / SeekBar.Maximum);
+                }
+            }
+            catch (Exception ex)
+            {
+                ShowErrorMessageBox(ex);
+            }
+            finally
+            {
+                SeekBarTimerEnabled = false;
+            }
+        }
+
+        private void UpdatePosition(float? position = null)
+        {
+            try
+            {
+                var mediaPlayer = MediaPlayer;
+                var pos = position ?? mediaPlayer?.Position ?? 0;
+                var timeSpanPosition = TimeSpan.FromMilliseconds((mediaPlayer?.Length ?? 0) * pos);
+                var elapsedTime = (timeSpanPosition - Position).TotalMilliseconds;
+                if (position != null && elapsedTime > 0 && elapsedTime < 750)
+                {
+                    return;
+                }
+
+                UpdateSeekBar(timeSpanPosition, pos);
+            }
+            catch (Exception ex)
+            {
+                ShowErrorMessageBox(ex);
+            }
+        }
+
+        private void UpdateTime(double? position = null)
```

Comment:
I don't understand this method, and how does it compare with `UpdatePosition`? How come doesn't it take a time as an argument? In my experience, time is quite always set, while position might be unknown (because length can be unknown). This can happen on live streams, and maybe on big files that are long to download?

Reply:
The UpdateTime method updates the **remaining** time label (the method should be renamed to UpdateRemainingTime). I can't show the remaining time without length and position (the time can't help in this case).
The UpdatePosition method is to update the seek bar position.
There are different methods because when the user moves the seek bar, I update the remaining time immediatly, and the stream position is not updated until the user has finished to move the seek bar.

Reply:
I think there are some big confusions here. `position` is used interchangeably with `time`.
Let's agree on naming:
- `position` is the position between 0 (begin) and 1 (end). It is a floating point value.
- `time` is the number of milliseconds elapsed since the beginning of the playback. It is a `long` and stored between 0 and `length`

line 688 : It is called with a `Position`. Chances are that with the sample we chose, the Length only changes when `Position == 0`, but this will not always be the case.
line 1295 : A `TotalMilliseconds` is passed
line 1207 : null is implicitely passed.

Then, below in this method, if `position` is not null, we assign it to `time` and is interpreted as a time.

I think we should also add comments on top of this method. I can see that the given position is nullable. How is `null` interpreted?

I think that `position` should be renamed to `time` and be set to a `long` value